Lib/fontTools/merge.py

```diff
@@ -564,6 +572,64 @@ def merge(self, m, tables):
 def mapLookups(self, lookupMap):
 	pass
 
+def getGlyphNameAndFontIndex(gid):
+	"""Gets glyph name and font index from the composite glyph id.
+
+	Args:
+		gid: string representing glyph id. E.g. uni0000#0
+
+	Returns:
+		A tuple with two elements. The first element is a string representing the 
+		glyph name. The second element is an integer representing the font index in 
+		command line args.
+	
+	Example:
+		'uni0000', 3 = getGlyphNameAndFontIndex('uni0000#3')
+	"""
+	assert '#' in gid and gid.split('#')[-1].isdigit() and len(gid.split('#')) == 2, 'incorrect gid format'
```

Comment:
why you assert `len(gid.split('#')) == 2`? I think `#` is a valid character in a glyph name so it could already be present. I think it's enough to split at the right-most '#', with `rsplit('#', 1)`.